application/src/main/java/com/softserve/lv460/application/service/ApplicationUserService.java

```diff
@@ -30,114 +30,112 @@
-                    jwtTokenProvider.generateRefreshToken(email)
-            );
-        }
-        throw new BadRefreshTokenException(REFRESH_TOKEN_NOT_VALID);
+  private final ApplicationUserRepository applicationUserRepository;
+  private final UserRegistrationRequestMapper userRegistrationRequestMapper;
+  private final JwtTokenProvider jwtTokenProvider;
+  private final PasswordEncoder passwordEncoder;
+  private final VerificationTokenRepository verificationTokenRepository;
+
+  public ApplicationUser save(UserRegistrationRequest userRequest) {
+    if (applicationUserRepository.existsByEmail(userRequest.getEmail())) {
+      log.error("User with email: " + userRequest.getEmail() + " already exist.");
+      throw new UserAlreadyRegisteredException(String.format(ErrorMessage.USER_ALREADY_EXISTS, userRequest.getEmail()));
     }
-
-
-    public JWTSuccessLogIn login(JWTUserRequest loginRequest, Authentication auth) {
-        ApplicationUser user = applicationUserRepository
-                .findByEmail(loginRequest.getEmail())
-                .orElseThrow(() -> new BadEmailException(String.format(ErrorMessage.USER_NOT_FOUND_BY_EMAIL,loginRequest.getEmail())));
-        return new JWTSuccessLogIn(user.getId(), jwtTokenProvider.generateAccessToken(auth), jwtTokenProvider.generateRefreshToken(auth), user.getFirstName());
+    ApplicationUser applicationUser = userRegistrationRequestMapper.toEntity(userRequest);
+    applicationUserRepository.save(applicationUser);
+    return applicationUser;
+
+  }
+
+  @Transactional
+  public JWTUserResponse updateAccessTokens(String refreshToken) {
+    String email;
+    try {
+      email = jwtTokenProvider.getEmailFromJWT(refreshToken);
+      log.info("Email from refreshToken: " + refreshToken + " is " + email);
+    } catch (ExpiredJwtException e) {
+      log.error("Bad Refresh Token: " + refreshToken);
+      throw new BadRefreshTokenException(REFRESH_TOKEN_NOT_VALID);
     }
-
-    public void changeUserPassword(Long userId, String password) {
-        ApplicationUser applicationUser = applicationUserRepository.findById(userId)
-                .orElseThrow(() -> new UsernameNotFoundException(String.format(ErrorMessage.USER_NOT_FOUND_BY_ID,userId)));
-        applicationUser.setPassword(passwordEncoder.encode(password));
-        applicationUserRepository.save(applicationUser);
+    ApplicationUser user = applicationUserRepository
+            .findByEmail(email)
+            .orElseThrow(() -> new BadEmailException(String.format(ErrorMessage.USER_NOT_FOUND_BY_EMAIL, email)));
+    log.info("User have found : " + user.toString());
+    String newRefreshTokenKey = jwtTokenProvider.generateRefreshToken(email);
+    log.info("New RefreshTokenKey : " + newRefreshTokenKey + " for email: " + email);
+    applicationUserRepository.updateSecret(newRefreshTokenKey, user.getId());
+    log.info("Updated secret : " + newRefreshTokenKey + " for id:" + user.getId());
```

Comment:
I'm not sure about so many logs, but your ":" have different position in every sentence